McCorrections/src/LeptonWeights.cpp

```diff
@@ -142,13 +145,18 @@ double LeptonWeights::GetTriggerEfficiency(EventInfoBase& eventInfo, bool isData
         if(electronSF.HasCrossTriggers() && std::abs(eventInfo.GetLeg(2).GetMomentum().eta()) < 2.1){
             const double ele_single_eff = electronSF.GetTriggerEff(eventInfo.GetLeg(1).GetMomentum(), isData);
             const double ele_cross_eff = electronSF.GetTriggerEffCross(eventInfo.GetLeg(1).GetMomentum(), isData);
-            const double tau_eff = tauTriggerWeight->GetEfficiency(channel, LorentzVectorM(eventInfo.GetLeg(2).GetMomentum()),
-                                                                   eventInfo.GetLeg(2)->gen_match(),
-                                                                   eventInfo.GetLeg(2)->decayMode(), tau_iso_wp, isData);
-           static const std::vector<std::string> triggerPaths_unPrescaled = {
-               "HLT_Ele35_WPTight_Gsf_v", "HLT_Ele24_eta2p1_WPTight_Gsf_LooseChargedIsoPFTau30_eta2p1_CrossL1_v" };
-           if(!eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_unPrescaled) &&
-               eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_eTau)) prescaled_weight = prescaled_weight_eTau;
+
+            double tau_eff = 1;
+            tau_eff = isData ?
+                tauTriggerWeight_eTau->SFProvider::getEfficiencyData(static_cast<float>(eventInfo.GetLeg(2).GetMomentum().pt()),
+                                                                        eventInfo.GetLeg(2)->decayMode()) :
+                tauTriggerWeight_eTau->SFProvider::getEfficiencyMC(static_cast<float>(eventInfo.GetLeg(2).GetMomentum().pt()),
+                                                                      eventInfo.GetLeg(2)->decayMode());
+
+            static const std::vector<std::string> triggerPaths_unPrescaled = {
+                "HLT_Ele35_WPTight_Gsf_v", "HLT_Ele24_eta2p1_WPTight_Gsf_LooseChargedIsoPFTau30_eta2p1_CrossL1_v" };
+            if(!eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_unPrescaled) &&
+                eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_eTau)) prescaled_weight = prescaled_weight_eTau;
```

Comment:
from where you take prescale weights? It should also depend on the period.

Reply:
This part I didn't touch, should be changed?

Reply:
If I remember correctly, this part was needed only for 2017, and it was written some time ago. Please, cross-check that for 2017 it correct and implement similar thing for 2016 and 2017 if needed.

Reply:
where should I take these values?

Reply:
It should be calculated using `brilcalc`. Actually, the way it is implemented now is not very reliable. Please, for the moment, keep it as-it-is for 2017, and it will be addressed in the future PRs.

Reply:
As discussed above, please, apply the prescale sf only for 2017